File: server/lib/database/import.dart
import 'dart:io';

import 'package:jinya_backup/database/export.dart';
import 'package:jinya_backup/database/models/backup_job.dart';

Future importData(ExportData exportData) async {
  for (final user in exportData.users) {
    try {
      await user.create(hash: false);
    } catch (exception) {
      stdout.writeln(exception.toString());
    }
  }
  for (final job in exportData.jobs) {
    try {
      await job.create();
    } catch (exception) {
      stdout.writeln(exception.toString());
    }
  }
  for (final backup in exportData.backups) {
    try {
      final job = await BackupJob.findByName(backup.job.name);
      backup.job = job;
      await backup.create();
    } catch (exception) {
      stdout.writeln(exception.toString());
    }
  }
}

File: server/database/import.go
package database

import (
	"context"
	"time"
)

func ImportDatabase(jobs []BackupJob, backups []StoredBackup) error {
	db, err := ConnectToDb()
	if err != nil {
		return err
	}

	tx, err := db.BeginTx(context.Background(), nil)
	if err != nil {
		return err
	}

	for _, job := range jobs {
		_, err = tx.Exec("INSERT INTO backup_job (id, name, host, port, type, username, password, remote_path, local_path) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)", job.Id, job.Name, job.Host, job.Port, job.Type, job.Username, job.Password, job.RemotePath, job.LocalPath)
		if err != nil {
			err = tx.Rollback()
			return err
		}
	}

	for _, backup := range backups {
		_, err = tx.Exec("INSERT INTO stored_backup (id, name, full_path, backup_date, backup_job_id) VALUES ($1, $2, $3, $4, $5)", backup.Id, backup.Name, backup.FullPath, backup.BackupDate.Format(time.RFC3339), backup.JobId)
		if err != nil {
			err = tx.Rollback()
			return err
		}
	}

	return tx.Commit()
}
